Python/Product/Cookiecutter/ViewModel/CookiecutterViewModel.cs

```diff
@@ -0,0 +1,726 @@
+                    _openFolder(OutputFolderPath);
+                }
+            } catch (ProcessException ex) {
+                IsCreating = false;
+                IsCreatingSuccess = false;
+                IsCreatingError = true;
+
+                _outputWindow.WriteLine(string.Join(Environment.NewLine, ex.Result.StandardOutputLines));
+                _outputWindow.WriteErrorLine(string.Join(Environment.NewLine, ex.Result.StandardErrorLines));
+
+                _outputWindow.WriteLine(string.Empty);
+                _outputWindow.WriteLine(string.Format(CultureInfo.CurrentUICulture, Strings.RunningTemplateFailed, SelectedTemplate.DisplayName));
+                _outputWindow.ShowAndActivate();
+            } catch (Exception ex) when (!ex.IsCriticalException()) {
+                IsCreating = false;
+                IsCreatingSuccess = false;
+                IsCreatingError = true;
+
+                _outputWindow.WriteErrorLine(ex.Message);
+
+                _outputWindow.WriteLine(string.Empty);
+                _outputWindow.WriteLine(string.Format(CultureInfo.CurrentUICulture, Strings.RunningTemplateFailed, SelectedTemplate.DisplayName));
+                _outputWindow.ShowAndActivate();
+            }
+        }
+
+        public void NavigateToGitHubHome() {
+            var url = SelectedTemplate?.GitHubHomeUrl;
+            if (url != null) {
+                Process.Start(url)?.Dispose();
+            }
+        }
+
+        public void NavigateToGitHubIssues() {
+            var url = SelectedTemplate?.GitHubIssuesUrl;
+            if (url != null) {
+                Process.Start(url)?.Dispose();
+            }
+        }
+
+        public void NavigateToGitHubWiki() {
+            var url = SelectedTemplate?.GitHubWikiUrl;
+            if (url != null) {
+                Process.Start(url)?.Dispose();
+            }
+        }
+
+        public void NavigateToHelp() {
+            Process.Start(UrlConstants.HelpUrl)?.Dispose();
+        }
+
+        public async Task SelectTemplate(TemplateViewModel template) {
+            SelectedTemplate = template;
+            await RefreshSelectedDescriptionAsync(template);
+        }
+
+        public async Task LoadMoreTemplates(string continuationToken) {
+            var last = GitHub.Templates.LastOrDefault();
+            if (last is ContinuationViewModel) {
+                GitHub.Templates.Remove(last);
+
+                await AddFromSource(_githubSource, null, KnownMonikers.GitNoColor, GitHub, continuationToken);
+            }
+        }
+
+        private async Task AddFromSource(ITemplateSource source, string searchTerm, VisualStudio.Imaging.Interop.ImageMoniker image, CategorizedViewModel parent, string continuationToken = null) {
+            var loading = new LoadingViewModel();
+            parent.Templates.Add(loading);
+
+            try {
+                var result = await source.GetTemplatesAsync(searchTerm, continuationToken, CancellationToken.None);
```

Comment:
>GetTemplatesAsync [](start = 42, length = 17)

I think we need to check the current search term after completing this method and abort if it has changed. Otherwise we'll get the race condition you were worried about.